internal/modules/nativeext/nativeext.go

```diff
@@ -205,9 +205,24 @@ func (h *Host) downloadFromRemote(ctx context.Context, source, name string, vers
 		return "", fmt.Errorf("failed to fetch release: %w", err)
 	}
 
-	bin, _, err := archive.Extract(ctx, fi.Name(), resp, archive.WithFilePath(filepath.Base(name)))
+	// TODO(jaredallard): The library should handle detecting this for us.
```

Comment:
Not going to merge anything until I sort out how I want to handle extension stacking.